agent/api/ecsclient/client.go

```diff
@@ -482,6 +482,31 @@ func (client *APIECSClient) SubmitContainerStateChange(change api.ContainerState
 	return nil
 }
 
+func (client *APIECSClient) SubmitAttachmentStateChange(change api.AttachmentStateChange) error {
+	if change.Attachment == nil { // this should not happen unless there's a bug
```

Comment:
If that's a bug, shouldn't you just panic and crash/stop the agent? Just logging the error message will hide the occurrence of that bug. If you do not want to invoke `panic` yourself, might be better to not do the check at all and have a NPE crashing the agent.

Reply:
removed this check